logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,228 @@
+  private final int mMinNumberOfThreads;
+  private final int mMaxNumberOfThreads;
+  private ExecutorService mThreadPool;
+  private volatile boolean mStopped;
+
+  /**
+   * Construct an {@link AlluxioLogServerProcess} instance.
+   *
+   * @param baseLogsDir base directory to store the logs pushed from remote Alluxio servers
+   */
+  public AlluxioLogServerProcess(String baseLogsDir) {
+    mPort = Configuration.getInt(PropertyKey.LOG_SERVER_PORT);
+    mMinNumberOfThreads = Configuration.getInt(PropertyKey.MASTER_WORKER_THREADS_MIN) + 2;
+    mMaxNumberOfThreads = Configuration.getInt(PropertyKey.MASTER_WORKER_THREADS_MAX) + 2;
+    mBaseLogsDir = baseLogsDir;
+    mStopped = true;
+  }
+
+  @Override
+  public void start() throws Exception {
+    startServing();
+  }
+
+  @Override
+  public void stop() throws Exception {
+    stopServing();
+  }
+
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mStopped == false;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  /**
+   * Create and start logging server and client thread pool.
+   */
+  private void startServing() {
+    SynchronousQueue<Runnable> synchronousQueue =
+        new SynchronousQueue<>();
+    mThreadPool =
+        new ThreadPoolExecutor(mMinNumberOfThreads, mMaxNumberOfThreads,
+            STOP_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS, synchronousQueue);
+    try {
+      mServerSocket = new ServerSocket(mPort);
+    } catch (IOException e) {
+      LOG.error("Failed to bind to port {}.", mPort);
+      return;
+    }
+    mStopped = false;
+    while (!mStopped) {
+      try {
+        Socket client = mServerSocket.accept();
+        InetAddress inetAddress = client.getInetAddress();
+        AlluxioLog4jSocketNode clientSocketNode =
+            new AlluxioLog4jSocketNode(this, client);
+        RetryPolicy retryPolicy = new ExponentialBackoffRetry(
+            BASE_SLEEP_TIME_IN_MS, MAX_SLEEP_TIME_IN_MS, MAX_NUM_RETRY);
+        while (true) {
+          try {
+            mThreadPool.execute(clientSocketNode);
+            break;
+          } catch (Throwable t) {
+            if (t instanceof RejectedExecutionException) {
+              if (!retryPolicy.attemptRetry()) {
+                LOG.warn("Connection with {} has been rejected by ExecutorService {} times"
+                    + "till timedout, reason: {}",
+                    inetAddress.getHostAddress(), retryPolicy.getRetryCount(), t);
+                client.close();
+                break;
+              }
+            } else if (t instanceof Error) {
+              LOG.error("ExecutorService threw error: ", t);
+              throw (Error) t;
+            } else {
+              LOG.warn("ExecutorService threw error: ", t);
+              break;
+            }
+          }
+        }
+      } catch (IOException e) {
```

Comment:
reduce the scope of this `try-catch` block so that we can have a better idea of which part actually failed

Reply:
Done.